### PR TITLE
Cleans up the Examples in the repository, and adds them to the Documentation

### PR DESCRIPTION
**Pull request by mdaneri**
### Changes:
- Listener moved to localhost and port 8081 when possible,
- Import-Module generate an exception if fails
- Cleanup 
- Tested
- Each Sample code include a `.SYNOPSIS` and `.DESCRIPTION`  
- New autogenerate document page with all sample code under /docs/Getting-Started
- Pode.Build.ps1 automatically generate the sample code summary page
- New Pester test to check that any sample code includes `.SYNOPSIS` and `.DESCRIPTION`   and `.NOTE`

## Header Example

```powershell
<#
.SYNOPSIS
    A sample PowerShell script to set up a Pode server and integrate with Redis for caching.
.DESCRIPTION
    This script sets up a Pode server listening on port 8081 with Redis integration for caching.
    It checks for the existence of the `redis-cli` command and sets up a Pode server with routes
    that demonstrate caching with Redis.
.NOTES
    Author: Pode Team
    License: MIT License
#>
```

### CODE REVIEW

File: examples/web-pages-docker.ps1 [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
 #>
 
-# create a server, and start listening on port 8085
+# create a server, and start listening on port 8081
 Start-PodeServer -Threads 2 {
 
-    # listen on *:8085
-    Add-PodeEndpoint -Address * -Port 8085 -Protocol Http
+    # listen on *:8081
+    Add-PodeEndpoint -Address * -Port 8081 -Protocol Http
```
Comment (Badgerati) [inline examples/web-pages-docker.ps1:14]: There's a `docker-compose.yml` and a `Dockerfile` in the examples folder which will need ports updating to 8081 as well :)

**Review by Badgerati — CHANGES_REQUESTED** [review]

**Comment by mdaneri** [comment]
I have not done yet with this. 
I plan to make a change in Pode.build.ps1 to automatically create an md document with any example with a brief description + a test to check if any example has the correct header for the build to create the page

**Comment by mdaneri** [comment]
.LINK is available on the summary page
.EXAMPLE do you think is needed?

File: pode.build.ps1 [hunk 2]
```diff
@@ -344,6 +344,97 @@ Task DocsDeps ChocoDeps, {
     Install-PodeBuildModule PlatyPS
 }
 
+Task IndexSamples {
```
Comment (Badgerati) [inline pode.build.ps1:347]: This task could be simplified to the following; which also fixes a few bugs with the original as well:
* Fixes bug with GitHub URLs missing subfolders
* Fixes issue with no extra space between multiple lines being concatenated
* Changed the page title to be just "Sample Scripts"
* The "Index" isn't needed in mkdocs, as mkdocs will make it automatically on the right

```powershell
Task IndexSamples {
    $examplesPath = './examples'
    if (!(Test-Path -PathType Container -Path $examplesPath)) {
        return
    }

    # List of directories to exclude
    $sampleMarkDownPath = './docs/Getting-Started/Samples.md'
    $excludeDirs = @('scripts', 'views', 'static', 'public', 'assets', 'timers', 'modules',
        'Authentication', 'certs', 'logs', 'relative', 'routes')

    # Convert exlusion list into single regex pattern for directory matching
    $dirSeparator = [IO.Path]::DirectorySeparatorChar
    $excludeDirs = "\$($dirSeparator)($($excludeDirs -join '|'))\$($dirSeparator)"

    # build the page content
    Get-ChildItem -Path $examplesPath -Filter *.ps1 -Recurse -File -Force |
        Where-Object {
            $_.FullName -inotmatch $excludeDirs
        } |
        Sort-Object -Property FullName |
        ForEach-Object {
            Write-Verbose "Processing Sample: $($_.FullName)"

            # get the script help
            $help = Get-Help -Name $_.FullName -ErrorAction Stop

            # add help content
            $urlFileName = ($_.FullName -isplit 'examples')[1].Trim('\/') -replace '[\\/]', '/'
            $markdownContent += "## [$($_.BaseName)](https://github.com/Badgerati/Pode/blob/develop/examples/$($urlFileName))`n`n"
            $markdownContent += "**Synopsis**`n`n$($help.Synopsis)`n`n"
            $markdownContent += "**Description**`n`n$($help.Description.Text)`n`n"
        }

    Write-Output "Write Markdown document for the sample files to $($sampleMarkDownPath)"
    Set-Content -Path $sampleMarkDownPath -Value "# Sample Scripts`n`n$($markdownContent)" -Force
}
```

File: tests/unit/_.Tests.ps1 [hunk 3]
```diff
@@ -1,5 +1,30 @@
 [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSUseDeclaredVarsMoreThanAssignments', '')]
 param()
+BeforeDiscovery {
```
Comment (Badgerati) [inline tests/unit/_.Tests.ps1:3]: Similar to the new Task, the BeforeDiscovery here can be simplified:

```powershell
BeforeDiscovery {
    $path = $PSCommandPath
    $examplesPath = (Split-Path -Parent -Path $path) -ireplace '[\\/]tests[\\/]unit', '/examples/'

    # List of directories to exclude
    $excludeDirs = @('scripts', 'views', 'static', 'public', 'assets', 'timers', 'modules',
        'Authentication', 'certs', 'logs', 'relative', 'routes')

    # Convert exlusion list into single regex pattern for directory matching
    $dirSeparator = [IO.Path]::DirectorySeparatorChar
    $excludeDirs = "\$($dirSeparator)($($excludeDirs -join '|'))\$($dirSeparator)"

    # get the example scripts
    $ps1Files = @(Get-ChildItem -Path $examplesPath -Filter *.ps1 -Recurse -File -Force |
        Where-Object {
            $_.FullName -inotmatch $excludeDirs
        }).FullName
}
```

File: examples/Web-Auth-Digest.ps1
Comment (Badgerati) [inline examples/Web-Auth-Digest.ps1:1]: Name needs changing to be `Web-AuthDigest`, to match the format of the others

File: examples/TcpServer-Auth.ps1
Comment (Badgerati) [inline examples/TcpServer-Auth.ps1:1]: Id change the name to be `Tcp-ServerAuth`, to match the format of the "Web" ones

File: examples/TcpServer-Http.ps1
Comment (Badgerati) [inline examples/TcpServer-Http.ps1:1]: Id change the name to be `Tcp-ServerHttp`, to match the format of the "Web" ones

File: examples/TcpServer-MultiEndpoint.ps1
Comment (Badgerati) [inline examples/TcpServer-MultiEndpoint.ps1:1]: Id change the name to be `Tcp-ServerMultiEndpoint`, to match the format of the "Web" ones

File: examples/Rest-Api.ps1
Comment (Badgerati) [inline examples/Rest-Api.ps1:1]: I'd change this to be `Web-RestApi.ps1`

File: examples/Nunit-RestApi.ps1
Comment (Badgerati) [inline examples/Nunit-RestApi.ps1:1]: I'd change this to be `Web-NunitRestApi.ps1`

File: examples/Web-Pages.ps1
Comment (Badgerati) [inline examples/Web-Pages.ps1:1]: There are two references to this file in `./examples/package.json`, which will need updating to be `./Web-Pages.ps1` with the new casing.

File: examples/Web-PagesDocker.ps1
Comment (Badgerati) [inline examples/Web-PagesDocker.ps1:1]: There's a reference to this file in `./examples/Dockerfile` which will need updating to be `./Web-PagesDocker.ps1`. The exposed port will also need updating to `8081` as well.

**Review by Badgerati — CHANGES_REQUESTED** [review]

File: examples/sse.ps1
[the remark this answers is not in the document]
Comment (mdaneri) [reply examples/sse.ps1:1]: This coming from the last merge. I don't see it on my Windows. Let me delete it from the repository

File: examples/Web-Pages.ps1
Comment (mdaneri) [reply examples/Web-Pages.ps1:1]: I missed it 
thanks

**Review by Badgerati — APPROVED** [review]

**Pull request merged**